File: String-Problems/Is-Rotation/Is-Rotation.py
def is_rotation(s, t):
    """
    Returns True iff s is a rotated version of t.
    """
    return len(s) == len(t) and s in t + t

if __name__ == "__main__":
    testStringPairs = [
        ("", ""),
        ("stackoverflow", "flowstackover"),
        ("12345", "5xy1234"),
        ("stackoverflaw", "overflowstack")
    ]
    for (index, pair) in enumerate(testStringPairs):
        s, t = pair
        print("Test String #{}:".format(index + 1))
        print("{} {} a rotated version of {}.".format(
            repr(s), "is" if is_rotation(s, t) else "is not", repr(t)
        ))

File: String-Problems/Is-Rotation/Is-Rotation.cs
using System;

public class IsRotationSolution
{
    public static bool IsRotation(string s, string t)
    {
        return s.Length == t.Length && (s + s).Contains(t);
    }

    /// <summary>
    /// The entry point.
    /// </summary>
    public static void Main()
    {
        var testStringPairs = new[]
        {
			new[] {"stackoverflow", "flowstackover"},
            new[] {"animal", "alanim"},
            new[] {"stackoverflow", "flowstackovez"},
            new[] {"stackoverflow", "flowstackove"}
		};
        var testCase = 1;
        foreach (var stringPair in testStringPairs)
        {
            var s = stringPair[0];
            var t = stringPair[1];
            Console.WriteLine("Test String Pair #{0}:", testCase);
            Console.WriteLine("({0}, {1}) {2} rotations of each other.",
                              s, t, IsRotation(s, t) ? "are" : "are not");
            testCase++;
        }
    }
}